ballerina/modules/parser/state.bal

```diff
@@ -62,8 +62,11 @@ public class ParserState {
 
     common:Event[] eventBuffer = [];
 
-    public isolated function init(string[] lines) returns ParsingError? {
+    boolean isString;
+
+    public isolated function init(string[] lines, boolean isString = false) returns ParsingError? {
```

Comment:
Could you please elaborate on the usage of this `isString` flag? 
can't we assume that all the lines fed into the parser as strings? not sure why we need to mark it explicitly

Reply:
In the `readFile` scenario, the input is separated by new lines, and the parser can determine this new line state by reaching the end of the given line. However, in the case of `readString`, the user is passing the entire string, and the parser has to determine the new lines through the newline character within the string itself. During the initial development, we had to read the lines as an array of strings; however, we now have the support to refactor the parser to consider a character stream instead of an array, which makes the parsing uniform for both APIs. This would be a separate improvement issue.